pkg/scheduler/dependency_graph.go

```diff
@@ -83,6 +92,35 @@ func newScheduledResourceFor(r interfaces.Resource, context *GraphContext) *Sche
 	}
 }
 
+type DependencyList client.DependencyList
+
+func (d *DependencyList) Len() int {
+	return len(d.Items)
+}
+
+func (d *DependencyList) Less(i, j int) bool {
+	if d.Items[i].CreationTimestamp.Equal(d.Items[j].CreationTimestamp) {
+		return d.Items[i].UID < d.Items[j].UID
+	}
+	return d.Items[i].CreationTimestamp.Before(d.Items[j].CreationTimestamp)
+}
+
+func (d *DependencyList) Swap(i, j int) {
```

Comment:
The goal was to sort list of dependencies by custom criteria. Since there are no generics in golang standard library function which does sort requires the collection to implement this particular interface with 3 methods. I wish I could provide it with comparator func, but I cannot.
There are two ways that I know to sort collection by custom criteria
1) Add those 3 methods to client package where the Dependency/Replica classes are declared since you cannot write struct methods in packages other than the package where the type was declared
2) Declare type alias (e.g. type ReplicaList client.ReplicaList), implement the sort interface for that type and then cast original object to that new type. This is a common technique in go.
I chose method 2 because I didn't feel like this methods belong in the client package since this is just an ad-hoc sorting and the sorting criteria might be different in other places